torch/_dynamo/variables/dicts.py

```diff
@@ -611,9 +627,17 @@ def unpack_var_sequence(self, tx):
     def reconstruct(self, codegen: "PyCodegen"):
         # load types.MappingProxyType
         if self.source:
-            unimplemented(
-                "Can't reconstruct an existing mapping variable because"
-                " the connection to the original dict will be lost"
+            msg = (
+                f"Preexisting MappingProxyVariable (source: {self.source}) cannot be reconstructed "
+                "because the connection to the original dict will be lost."
+            )
+            unimplemented_v2(
+                gb_type="mapping proxy cannot be reconstructed",
+                context=f"Source: {self.source}",
+                explanation=msg,
+                hints=[
+                    "Use a mapping proxy constructed in the same `torch.compile` region."
+                ],
```

Comment:
Add `graph_break_hints.SUPPORTABLE` as well. This graph break is very bizarre...